### PR TITLE
[coqdoc] Add links for existing instances

### PR DESCRIPTION
**Pull request by proux01**
Dump in glob file references for `Existing Instance` and `Existing Class` to make coqdoc display an hyperlink to their definitions.

- [x] Added / updated **test-suite**.

- [ ] Added **changelog**.
- ~[ ] Added / updated **documentation**.~
  
  - ~[ ] Documented any new / changed **user messages**.~
  - ~[ ] Updated **documented syntax** by running `make doc_gram_rsts`.~

- [ ] Opened **overlay** pull requests.

<!-- Pointers to relevant developer documentation:

Contributing guide: https://github.com/coq/coq/blob/master/CONTRIBUTING.md

Test-suite: https://github.com/coq/coq/blob/master/test-suite/README.md

Changelog: https://github.com/coq/coq/blob/master/doc/changelog/README.md

Building the doc: https://github.com/coq/coq/blob/master/doc/README.md
Sphinx: https://github.com/coq/coq/blob/master/doc/sphinx/README.rst
doc_gram: https://github.com/coq/coq/blob/master/doc/tools/docgram/README.md

Overlays: https://github.com/coq/coq/blob/master/dev/ci/user-overlays/README.md

### CODE REVIEW

File: vernac/classes.ml [hunk 1]
```diff
@@ -276,7 +275,7 @@ let existing_instance glob g info =
   let ctx, r = Term.decompose_prod_decls instance in
     match class_of_constr (Environ.push_rel_context ctx env) sigma (EConstr.of_constr r) with
       | Some (_, ((tc,u), _)) -> add_instance tc info glob c
-      | None -> user_err ?loc:g.CAst.loc
+      | None -> user_err
```
Comment (SkySkimmer) [inline vernac/classes.ml:278]: Why drop the loc?

**Comment by SkySkimmer** [comment]
This is more than "Add links for existing instances", it changes the grammar.

File: vernac/classes.ml (hunk 1 above)
Comment (proux01) [reply vernac/classes.ml:278]: Because it is not needed (and no longer available in the globref).

**Comment by proux01** [comment]
> This is more than "Add links for existing instances", it changes the grammar.

Sure, but it is a conservative extension (simply enabling to give references through notations).

File: vernac/classes.ml (hunk 1 above)
Comment (SkySkimmer) [reply vernac/classes.ml:278]: Locs are never needed but it's still nice to be precise.

>no longer available in the globref

so add a ?loc argument

**Comment by SkySkimmer** [comment]
Sure, but PR titles and commit messages should try to be accurate

**Comment by proux01** [comment]
Anyway, the change in grammar was not the one I thought and was really strange in fact (who would like to provide existing instances with a string?) so here is a simpler solution not modifying the grammar. Thanks for pointing this out!

**Review by silene — APPROVED** [review]
Do we want to add/modify a test in `test-suite/coqdoc`? Other than that, looks good to me.

**Comment by proux01** [comment]
Indeed, done

**Comment by jfehrle** [comment]
Hmm, the `coqdoc` documentation needs a good bit of work.  There should be a few screen shots of the output and instructions on how to run it on your project.  Also the `glob` file format should probably be described.  But not in this PR.

**Comment by Alizter** [comment]
The glob file format is at least documented in the source code: https://github.com/coq/coq/pull/14966/files

**Comment by silene** [comment]
This does not look like a proper test-suite file. Make sure it was generated with
```shell
coqdoc -utf8 -R . Coqdoc -coqlib_url http://coq.inria.fr/stdlib --html foo.v
```
and same thing with `--latex`.

(This example also shows how bad attributes are handled, but that is an issue for another time.)

**Comment by proux01** [comment]
Thansks, fixed.
I was mislead by the fact that `make coqdoc` in the test-suite didn't display any failure. Failures are only visible in the `*.v.log` files. Isn't that a bug? (note that the CI here was green despite the failure)

**Comment by silene** [comment]
The fact that `make foo` only produces log files and does not report errors is a feature, I think. You are supposed to run `make report` to get the actual result. The fact that the test-suite succeeded, however, is a bug. In fact, it correctly detected the failure, and then immediately forgot about it:
```
==========> FAILURE <==========
    coqdoc/typeclasses.v...Error! (unexpected output)
FAILED    coqdoc/typeclasses.v.log
NO FAILURES
make[1]: Leaving directory '/builds/coq/coq/test-suite'
Cleaning up project directory and file based variables 00:00
Job succeeded
```
This is worrying.

**Comment by proux01** [comment]
> The fact that `make foo` only produces log files and does not report errors is a feature, I think. You are supposed to run `make report` to get the actual result.

Why not, although this is not consistent with other directories in `test-suite` (for instance `make output` displays failures on the fly and I never had to run report there).

**Comment by silene** [comment]
@coqbot run full ci

**Comment by proux01** [comment]
@silene I rebased to enjoy your Makefile fix from https://github.com/coq/coq/pull/17257 and CI remains green, so we should be all set now.

**Comment by silene** [comment]
@coqbot merge now

**Pull request merged**